HEN_HOUSE/egs++/egs_application.cpp

```diff
@@ -269,6 +269,13 @@ EGS_Application::EGS_Application(int argc, char **argv) : input(0), geometry(0),
     sprintf(buf,"egsrun_%d_",egsGetPid());
     run_dir = buf;
     if (input_file.size() > 0) {
+
+        // Remove the .egsinp extension if it was included
+        size_t ext = input_file.find_last_of(".egsinp");
```

Comment:
The search function should be `input_file.rfind(".egsinp")`, and then below `input_file = input_file.substr(0,ext)`; `find_last_of` searches the string for the last occurrence of any one of the characters in the character set `.egsinp`. The following code returns `myin` because the last character `e` is matched:
```
int main()
{    
    std::string str = "myinputfile";
    std::size_t ext = str.find_last_of(".egsinp");
    str = str.substr(0, ext-6);
    std::cout << str << "\n";
}
```